src/screens/scanning/VoucherEntryNavigator.tsx

```diff
@@ -0,0 +1,128 @@
+import Colors from '../../../assets/Colors';
+import { useScanningContext } from './ScanningContext';
+import { handlePreventLeave } from '../../utils/scanningUtils';
+import styles, { NavButton, NavButtonContainer } from './styles';
+import ManualVoucherScreen from './ManualVoucherScreen';
+import ScanningScreen from './ScanningScreen';
+import VoucherBatchScreen from './VoucherBatchScreen';
+
+export default function VoucherEntryNavigator({
+  navigation,
+}: ScannerStackScreenProps<'VoucherEntryNavigator'>) {
+  const [selection, setSelection] = useState(1);
+  const { voucherMap, dispatch } = useScanningContext();
+  const hasUnsavedChanges = Boolean(voucherMap.size);
+
+  return (
+    // this line enables the keyboard to be dismissed when the user taps anywhere on the screen
+    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+      <SafeArea>
+        <StandardHeader width="80%">
+          <TitleText>Add a voucher</TitleText>
+          <TouchableOpacity
+            onPress={() =>
+              handlePreventLeave({
+                hasUnsavedChanges,
+                navigation,
+                dispatch,
+              })
+            }
+          >
+            <Icon name="close" size={24} color={Colors.midBlack} />
+          </TouchableOpacity>
+        </StandardHeader>
+        <View>
+          <NavButtonContainer>
+            <NavButton
+              style={
+                selection === 1 ? styles.selectedBtn : styles.unselectedBtn
+              }
+              onPress={() => {
+                setSelection(1);
+              }}
+            >
+              <NavButtonText
+                style={
+                  selection === 1
+                    ? { color: Colors.magenta }
+                    : { color: Colors.midGray }
+                }
+              >
+                Serial
+              </NavButtonText>
+            </NavButton>
+            <NavButton
+              style={
+                selection === 2 ? styles.selectedBtn : styles.unselectedBtn
+              }
+              onPress={() => {
+                setSelection(2);
+              }}
+            >
+              <NavButtonText
+                style={
+                  selection === 2
+                    ? { color: Colors.magenta }
+                    : { color: Colors.midGray }
+                }
+              >
+                Range
+              </NavButtonText>
+            </NavButton>
+            <NavButton
+              style={
+                selection === 3 ? styles.selectedBtn : styles.unselectedBtn
+              }
+              onPress={() => {
+                setSelection(3);
+              }}
+            >
+              <NavButtonText
+                style={
+                  selection === 3
+                    ? { color: Colors.magenta }
+                    : { color: Colors.midGray }
+                }
+              >
+                Barcode
+              </NavButtonText>
+            </NavButton>
```

Comment:
Maybe make a NavButton component that takes in the button text as a prop. Removes repeated code.